test/e2e/storage/regional_pd.go

```diff
@@ -0,0 +1,438 @@
+	claimTemplate := newClaimTemplate(ns)
+	claimTemplate.Spec.StorageClassName = &class.Name
+	statefulSet, service, repdLabels := newStatefulSet(claimTemplate, ns)
+
+	By("creating a StorageClass " + class.Name)
+	_, err := c.StorageV1().StorageClasses().Create(class)
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		framework.Logf("deleting storage class %s", class.Name)
+		framework.ExpectNoError(c.StorageV1().StorageClasses().Delete(class.Name, nil),
+			"Error deleting StorageClass %s", class.Name)
+	}()
+
+	By("creating a StatefulSet")
+	_, err = c.CoreV1().Services(ns).Create(service)
+	Expect(err).NotTo(HaveOccurred())
+	_, err = c.AppsV1().StatefulSets(ns).Create(statefulSet)
+	Expect(err).NotTo(HaveOccurred())
+
+	err = framework.WaitForStatefulSetReplicasReady(statefulSet.Name, ns, c, framework.Poll, statefulSetReadyTimeout)
+	if err != nil {
+		pod := getPod(c, ns, repdLabels)
+		Expect(podutil.IsPodReadyConditionTrue(pod.Status)).To(BeTrue(),
+			"The statefulset pod has the following conditions: %s", pod.Status.Conditions)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	pvc := getPVC(c, ns, repdLabels)
+	defer func() {
+		framework.Logf("deleting statefulset%q/%q", statefulSet.Namespace, statefulSet.Name)
+		// typically this claim has already been deleted
+		framework.ExpectNoError(c.AppsV1().StatefulSets(ns).Delete(statefulSet.Name, nil /* options */),
+			"Error deleting StatefulSet %s", statefulSet.Name)
+		framework.Logf("deleting claim %q/%q", pvc.Namespace, pvc.Name)
+		framework.ExpectNoError(c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Delete(pvc.Name, nil),
+			"Error deleting claim %s.", pvc.Name)
+		err = framework.WaitForPersistentVolumeDeleted(c, pvc.Spec.VolumeName, framework.Poll, pvDeletionTimeout)
+		if err != nil {
+			framework.Logf("WARNING: PV %s is not yet deleted, and subsequent tests may be affected.", pvc.Spec.VolumeName)
+		}
+	}()
+
+	By("getting zone information from pod")
+	pod := getPod(c, ns, repdLabels)
+	nodeName := pod.Spec.NodeName
+	node, err := c.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+	podZone := node.Labels[apis.LabelZoneFailureDomain]
+
+	// TODO (verult) Consider using node taints to simulate zonal failure instead.
+	By("deleting instance group belonging to pod's zone")
+
+	// Asynchronously detect a pod reschedule is triggered during/after instance group deletion.
+	waitStatus := make(chan error)
+	go func() {
+		waitStatus <- waitForStatefulSetReplicasNotReady(statefulSet.Name, ns, c)
+	}()
+
+	cloud, err := framework.GetGCECloud()
+	if err != nil {
+		Expect(err).NotTo(HaveOccurred())
+	}
+	instanceGroupName := framework.TestContext.CloudConfig.NodeInstanceGroup
+	instanceGroup, err := cloud.GetInstanceGroup(instanceGroupName, podZone)
+	Expect(err).NotTo(HaveOccurred(),
+		"Error getting instance group %s in zone %s", instanceGroupName, podZone)
+	err = framework.DeleteManagedInstanceGroup(podZone)
```

Comment:
You already have the instanceGroupName here, could you just pass that in, instead of the zone name?

Reply:
DeleteManagedInstanceGroup() already has the instance group name (from the TestContext), but the same name is used in every zone, so it needs the zone name to identify the right instance group.

Reply:
We do, because it's used to get the instance group through the cloud provider API, which has to happen before deleting the group.